examples/CayenneLPP/CayenneLPP.ino

```diff
@@ -79,24 +112,46 @@ void sleep()
 
 void onMotionStart()
 {
-  node->setColor(TTN_BLUE);
+  node->setColor(TTN_RED);
 
   debugSerial.print("-- SEND: MOTION");
   sendData(PORT_MOTION);
 }
 
 void onButtonRelease(unsigned long duration)
 {
-  node->setColor(TTN_BLUE);
-
-  debugSerial.print("-- SEND: BUTTON");
-  debugSerial.println(duration);
+  uint32_t timepressed = (uint32_t) duration;
+
+  snprintf_P(buf, sizeof(buf), PSTR("-- SEND: BUTTON %d ms"), timepressed);
+  debugSerial.println(buf);
+
+  // If button was pressed for more then 2 seconds
+  if (timepressed > 2000) {
+    // blink yellow led for 60 seconds
+    // this will let us to upload new sketch if needed
+    for (uint8_t i=0 ; i<60 ; i++) {
+      node->setColor(TTN_YELLOW);
+      delay(30);
+      node->setColor(TTN_BLACK);
+      delay(470);
+    }
+  }
 
-  sendData(PORT_BUTTON);
+  // then send data
+  sendData(PORT_BUTTON, timepressed);
 }
 
-void sendData(uint8_t port)
+void sendData(uint8_t port, uint32_t duration)
 {
+  uint16_t bat = node->getBattery();
+
+  // when battery < 3.3V (regulator does not regule output, so 
+  // we use another method to read VCC of the device and this will 
+  // also avoid error if VCC <  ADC reference (2.52V)
+  if (bat <= 3300) 
+  {
+    bat = node->getVcc();
+  }
```

Comment:
Shouldn't `getBattery()` do this?

Reply:
Yes, but `getBattery()` fails under 2.5V because ADC reference is 2.5V so in this case we use another technique that use the 1.1V ADC reference voltage and measure the Atmel VCC pin
see https://www.thethingsnetwork.org/forum/t/the-things-node-whats-this-3-2ma-sleep-mode-set-to-30ua-now/12521/74

Reply:
Yes but I mean should't `getBattery()` implement this fallback itself?

Reply:
Absolutely, it's in my list ;-)
But I'm under testing the difference for now in sketch BatteryMonitorLPP to see difference, so I need to have your original battery measurement to see differences.